nemo/collections/asr/models/wav2vec_models.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Optional
+
+import torch
+from omegaconf import DictConfig
+from pytorch_lightning import Trainer
+from torch import nn
+
+from nemo.collections.asr.models.ssl_models import SpeechEncDecSelfSupervisedModel
+from nemo.core.classes.common import PretrainedModelInfo, typecheck
+
+
+class Wav2VecEncoderModel(SpeechEncDecSelfSupervisedModel):
```

Comment:
Do we actually need this class? Maybe we could just use the parent class with additional optional features added to it, and/or to contrastive loss?

Reply:
Good point. Any reason we don't keep a simple linear decoder lying around though?

Reply:
It depends, the main differences are that the parent class detatches spectrograms, the child class has some extra dropouts here and there, and there's an additional penalty added to the loss. It's easy to add but would be only for the sake of only one model.

Reply:
I think these are minor enough to be options in that class, and we will probably experiment with them for other models later on anyway.

Reply:
Alrighty, I'll add them in